tests/run_tests.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+NUM_FAILED=0
+
+# Determine path of the current script
+HERE=$(cd `dirname $0` && pwd)
+RESOURCES=$(echo "${HERE}/.." | sed "s,/tests/..,,")
+
+# Determine paths to endless-sky executable and other relevant data
+ES_EXEC_PATH="${RESOURCES}/endless-sky"
+ES_CONFIG_TEMPLATE_PATH="${RESOURCES}/tests/config"
+
+echo "TAP version 13"
+echo "# ***********************************************"
+echo "# ***         ES Autotest-runner              ***"
+echo "# ***********************************************"
+echo "# Using Test Anything Protocol for reporting test-results"
+
+if [ ! -f "${ES_EXEC_PATH}" ]
+then
+	echo "1..1"
+	echo "not ok 1 Endless sky executable not found."
+	exit 1
+fi
+
+if [ ! -x "${ES_EXEC_PATH}" ]
+then
+	echo "1..1"
+	echo "not ok 1 Endless sky executable not executable. (did you use artifact downloading?)"
+	exit 1
+fi
+
+
+TESTS=$("${ES_EXEC_PATH}" --tests --resources "${RESOURCES}")
+if [ $? -ne 0 ]
+then
+	echo "1..1"
+	echo "not ok 1 Could not retrieve testcases"
+	exit 1
+fi
+TESTS_OK=$(echo "${TESTS}" | grep -e "^ACTIVE" | cut -d$'\t' -f2)
+TESTS_NOK=$(echo "${TESTS}" | grep -e "^KNOWN FAILURE" -e "^MISSING FEATURE" | cut -d$'\t' -f2)
+TESTS_TOTAL=$(( 0 + $(echo "${TEST_OK}" | wc -l) ))
+
+#TODO: Allow running known-failures by default as well (to check if they accidentally got solved)
+if [ ${TESTS_TOTAL} -eq 0 ]
+then
+	echo "1..1"
+	echo "not ok 1 Could not find any testcases"
+	exit 1
+fi
+
+echo "1..${TESTS_TOTAL}"
+
+# Set separator to newline (in case tests have spaces in their name)
+IFS_OLD=${IFS}
+IFS="
+"
+
+# Run all the tests
+for TEST in ${TESTS_OK}
+do
+	# Setup environment for the test
+	ES_CONFIG_PATH=$(mktemp --directory)
+	if [ ! $? ]
+	then
+		echo "not ok Couldn't create temporary directory"
+		echo "Bail out! Serious storage issue if we cannot create a temporary directory."
+		exit 1
+	fi
+
+	ES_SAVES_PATH="${ES_CONFIG_PATH}/saves"
+	mkdir -p "${ES_CONFIG_PATH}"
+	mkdir -p "${ES_SAVES_PATH}"
+	cp ${ES_CONFIG_TEMPLATE_PATH}/* ${ES_CONFIG_PATH}
+
+	TEST_NAME=$(echo ${TEST} | sed "s/\"//g")
+	TEST_RESULT="ok"
+	echo "# Running test ${TEST} with ${ES_CONFIG_PATH}"
+	# Use pipefail and use sed to remove ALSA messages that appear due to missing soundcards in the CI environment
+	set -o pipefail
+	"$ES_EXEC_PATH" --resources "${RESOURCES}" --test "${TEST_NAME}" --config "${ES_CONFIG_PATH}" 2>&1 |\
+		sed -e "/^ALSA lib.*$/d" -e "/^AL lib.*$/d" | sed "s/^/# /"
+	if [ $? -ne 0 ]
+	then
+		if [ -f "${ES_CONFIG_PATH}/errors.txt" ]
+		then
+			echo "# errors.txt:"
+			cat "${ES_CONFIG_PATH}/errors.txt" | sed "s/^/# /"
+		fi
+		TEST_RESULT="not ok"
+		NUM_FAILED=$((NUM_FAILED + 1))
+	fi
+	echo "${TEST_RESULT} ${TEST}"
```

Comment:
We should include the test number here, e.g.
![image](https://user-images.githubusercontent.com/20871346/100814499-69d64e80-3407-11eb-87b1-ee70e02283e4.png)